Mathlib/RingTheory/Ideal/Pointwise.lean

```diff
@@ -0,0 +1,115 @@
+/-
+Copyright (c) 2024 Eric Wieser. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Wieser
+-/
+import Mathlib.RingTheory.Ideal.Maps
+
+
+/-! # Pointwise instances on `Ideal`s
+
+This file provides the action `Ideal.pointwiseMulAction` which morally matches the action of
+`mulActionSet` (though here an extra `Ideal.span` is inserted).
+
+This actions is available in the `Pointwise` locale.
+
+## Implementation notes
+
+This file is similar (but not identical) to `RingTheory/Subring/Pointwise.lean`.
+Where possible, try to keep them in sync.
```

Comment:
This file doesn't seem to exist. Do you mean `Algebra/Ring/Subsemiring/Pointwise.lean`? I was going to say "you might want to put a note in that file suggesting that it be kept in sync with this one", but then I looked at that file and there's already a note saying that it should be kept in sync with a third file, which also doesn't seem to exist.

Reply:
I think @YaelDillies move all the files around, but the comments were left with the old names.

Reply:
My regex assumed files were referred to as `A.B.C`, not `A/B/C` 😐

Reply:
OK but just to be clear: purely for the purposes of reviewing this PR, which file am I supposed to be checking that it's similar to?

Reply:
Wherever you see a missing file, try replacing `RingTheory` by `Algebra.Ring`

Reply:
And `GroupTheory/Submonoid/Pointwise.lean` (mentioned in the subsemiring file) -> `Algebra/Group/Submonoid/Pointwise.lean` presumably?

Reply:
#13469

Reply:
So should the file in this PR be called `Mathlib/Algebra/Ring/Ideal/Pointwise.lean`?

Reply:
Perhaps it should be called that in a future PR, but for now the important thing is that it live next to `RingTheory/Ideal/Basic.lean`, which is where I have it now. If @YaelDillies  wants to move all of the files somewhere else in a later PR I'm happy to stay out of their way!